tests/readrecursive.c

```diff
@@ -68,7 +68,7 @@ static int readdirectory(char *dirname, int *dirsread, int *filesread, int *erro
         else {
             int bytes_read;
             char rbuf [1025];
-            fd = open(fn, O_RDWR);
+            fd = open(fn, O_RDONLY);
```

Comment:
This is a test file. I was considering adding O_RDWR to the list of 'write' flags to open to cause it to fail if in readonly mode, then realized that without thinking I used O_RDWR to open a file which was merely for reading. So I changed to test to be O_RDONLY so it could be used to test read success while testing readonly mode. Then I decided not to add O_RDWR to the 'write' flags. Still, the most correct way to open a file which will merely be read from is to use O_RDONLY.
This test just recurses through the filesystem, enters all directories, and reads all files.